kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -254,6 +274,44 @@ private AtomicWriteResult flush(final long offset, final UUID txnid) {
     );
     buffer.clear();
 
+    maybeTruncateChangelog(offset);
+    return AtomicWriteResult.success(partition);
+  }
+
+  private AtomicWriteResult drain(
+      final Iterator<CompletionStage<AtomicWriteResult>> it
+  ) {
+    try {
+      final var inFlight = new ArrayList<CompletableFuture<AtomicWriteResult>>();
+
+      while (it.hasNext()) {
+        inFlight.add(it.next().toCompletableFuture());
+        if (inFlight.size() >= ResponsiveConfig.MAX_CONCURRENT_REMOTE_WRITES_DEFAULT) {
+          for (final CompletableFuture<AtomicWriteResult> fut : inFlight) {
+            final var awr = fut.get();
```

Comment:
we don't need to block on all of these right? It's probably a bit faster to use something like `anyOf` over all the in-flight requests, clear the ones that are complete once `get` returns, and send out as many in-flight requests as we cleared.

Reply:
yeah, I was thinking of making that change but decided to put it off for a later optimization. I can make it now

Reply:
I think it'll be worthwhile. This is where tail latencies really start to bite.